queries.yaml

```diff
@@ -1122,19 +1122,47 @@ post_process:
       start_zoom: 4
       end_zoom: 13
 
+  # first, merge linestrings together where the properties are the same, and
+  # make sure we merge across junctions. this will create a set of non-simple
+  # multilinestrings which do not have nodes at junctions.
+  #
+  # note that the linestrings are already simplified by preceding simplify
+  # and clip stage.
   - fn: vectordatasource.transform.merge_line_features
     params:
       source_layer: roads
       start_zoom: 8
       end_zoom: 16
-
+      # setting the following will try to merge linestrings across junctions
+      # (i.e: more than 2 roads meeting at a point) where the angle between
+      # roads at that point is less than 5 degrees.
+      merge_junctions: true
+      merge_junction_angle: 5.0
+
+  # simplify roads again, to take advantage of any opportunities opened up
+  # by merging roads with the same properties in the previous step.
   - fn: vectordatasource.transform.simplify_layer
     params:
       source_layer: roads
       start_zoom: 8
       end_zoom: 16
       tolerance: 1.0
 
+  # merge _again_, this time to merge any features which no longer intersect
+  # because of the simplification step and can be packed into the same
+  # MultiLineString. also, drop short segments within the MultiLineString
+  # which didn't get merged into something larger - at 0.1 pixels, these
+  # probably aren't visible anyway.
+  - fn: vectordatasource.transform.merge_line_features
+    params:
+      source_layer: roads
+      start_zoom: 8
+      end_zoom: 16
+      # setting the following will cause lines, or parts of multi-lines,
+      # shorter than 0.1px at nominal zoom to be dropped.
+      drop_short_segments: true
+      drop_length_pixels: 0.1
```

Comment:
Thanks for making this configurable. Did you try a larger value like 0.2 or 0.5?

Reply:
The issue that I saw in a few places was that of very small segments, often less than a (Mercator) metre. These seem to have been created by tiny gaps between adjacent landuse areas (which give the roads a `landuse_kind`, meaning they're merged separately).

Getting rid of these almost-points probably doesn't affect the appearance of the tile in this case, but I'm worried that it might in other places. I think if we find we have more substantial fragments of road, we might want to try either dropping the attributes which are preventing them from being merged, or altering the algorithms (e.g: intercut) so as not to create them in the first place.

Otherwise, we risk creating little gaps in the road network, and in some places where the attributes change in a short distance (e.g: bridge, bus routes, cycleways, surface, etc...) these little gaps might add up to something visible.

Reply:
Sounds reasonable. Merge ahead!